### PR TITLE
tS3Connection setPathStyleAccess for better MinIO support

### PR DESCRIPTION
**Pull request by bgunics-talend**
Added a checkbox under Advanced settings of tS3Connection when a custom endpoint is used to enable pathstyle access.

When MinIO is used via DNS/hostname this must be checked otherwise the operations fail.

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-44045

**What is the new behavior?**
https://jira.talendforge.org/browse/TDI-44045

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by dmytro-sylaiev** [comment]
Hello @bgunics-talend 
You have added checkbox only for tS3Connection, but I think also need to add it for every tS3XXX component when not use the existing connection
I'll do that in separate PR

**Comment by bgunics-talend** [comment]
Hello @dmytro-sylaiev  unfortunately that would require a lot of change in all the other components. The custom region endpoint seems to be only available in the tS3Connection component. 
For example compare the tS3Get and tS3Connection advanced setting.

File: main/plugins/org.talend.designer.components.localprovider/components/tS3Connection/S3Client.javajet [hunk 1]
```diff
@@ -408,6 +412,9 @@
                     <%}%>
                     .withCredentials(credentialsProvider_<%=cid%>)
                     .withClientConfiguration(cc_<%=cid%>)
+					<%if(enablePathStyleAccess) {%>
+					.enablePathStyleAccess()
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tS3Connection/S3Client.javajet:416]: When **USE_REGION_ENDPOINT** is **unchecked** and **PATHSTYLEACCESS** is **checked**.
This should not be generated. but currently I think the **.enablePathStyleAccess()** would be generated.
Please check this.

Comment (bgunics-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tS3Connection/S3Client.javajet:416]: I've added the useRegionEndpoint this way PathStyle is ignored if the custom region is not set.

**Pull request merged**